### ISSUE TITLE
Before Connecting hook for Remote SSH

### ISSUE THREAD

**Issue by awschristou** [microsoft/vscode-remote-release#2775]
We encountered an interesting [scenario](https://github.com/aws/aws-toolkit-vscode/issues/1011) where it could be useful to provide users with a hook to perform some pre-connection operations before making a Remote SSH connection.

Some ideas:
* surface a hook from VS Code settings (something like `remote.SSH.onWillConnect`) which is executed prior to making the connection, when configured
* a command that other extensions can call (using vscode.commands.executeCommand) to wrap the functionality and provide custom before (and maybe after) handling

**Comment by josfaber** [comment]
Great! And while you're at it, a remote.SSH.onConnected would be awesome too ;-)

**Comment by roblourens** [comment]
This isn't really possible for us to do, given the way that extensions and remote resolvers work in vscode. There is another issue for the request to run a custom script before starting the remote server, which might cover whatever you have in mind.

**Comment by justinmk3** [comment]
> There is another issue for the request to run a custom script before starting the remote server

Is that one of these?

- https://github.com/microsoft/vscode-remote-release/issues/141
- https://github.com/microsoft/vscode-remote-release/issues/1722
